src/zip/mod.rs

```diff
@@ -982,6 +982,29 @@ macro_rules! map_impl {
                     dimension: self.dimension,
                 }
             }
+
+            /// Apply and collect the results into a new array, which has the same size as the
+            /// inputs.
+            ///
+            /// If all inputs are c- or f-order respectively, that is preserved in the output.
+            ///
+            /// Restricted to functions that produce copyable results for technical reasons; other
+            /// cases are not yet implemented.
+            pub fn apply_collect<R>(self, mut f: impl FnMut($($p::Item,)* ) -> R) -> Array<R, D>
+                where R: Copy,
+            {
+                unsafe {
+                    let is_c = self.layout.is(CORDER);
+                    let is_f = !is_c && self.layout.is(FORDER);
+                    let mut output = Array::uninitialized(self.dimension.clone().set_f(is_f));
```

Comment:
If this is considered UB - possibly, see comments in #685 - then a separate buffer type needs to be implemented.

Reply:
It looks like the problematic part remaining with this is `Vec::drop` which uses effectively `&mut self[..]`, and this code needs to be revisited.